File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_array  = list(map(lambda x: [int(i) for i in  x.split(' ')] , matrix_string.split('\n')))
    

    def matrix(self):
        return self.matrix_array

    def row(self, index):
        return self.matrix_array[index-1]

    def column(self, index):
        return [i[index-1] for i in self.matrix_array]

File: javascript/matrix/matrix.js
export class Matrix {
  constructor(inputString) {
    const getRow =  row => row.split(' ').map(t => parseInt(t))
    const getCol =  (matrix,colID) => matrix.map(x=>x[colID])
    const allRows = inputString.split('\n').map(x => getRow(x)) 
    this.matrixcols =  Object.keys(allRows[0]).map(x => getCol(allRows,x))
    this.matrixrows = allRows
  }

  get rows() {
    return this.matrixrows
  }

  get columns() {
    return this.matrixcols
  }
}
